src/teehr/metrics/gumboot_bootstrap.py

```diff
@@ -0,0 +1,185 @@
+"""Bootstrap classes."""
+from typing import Any, Callable, cast, Union
+from collections.abc import Generator as PyGenerator
+from pathlib import Path
+import logging
+
+from arch.bootstrap import IIDBootstrap
+from arch.bootstrap.base import (
+    # _loo_jackknife, # leave one out jackknife
+    _add_extra_kwargs
+)
+from arch.typing import ArrayLike, Int64Array, Float64Array, NDArray
+from numpy.random import Generator, RandomState
+import numpy as np
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+# FILL_VALUE = -9999  # JAB
+
+
+class GumbootBootstrap(IIDBootstrap):
```

Comment:
Not sure where this comment really belongs, but the quantiles returned by GumBoots are in a different order than the other bootstrap methods.  So if quantiles [0.05, 0.95] are requested, the results come back in a different order.  This is not a huge deal, but if it is easy to address, I think we should.

Reply:
Maybe the Struct contents are not ordered, so random?  The order they are requested seems to make no difference.

Reply:
Not sure how to maintain the order of the input quantiles in the output.  I'm actually using `MapType` rather than `StructType`, not sure if that makes a difference.  From some googling it seems that the order gets jumbled due to partitioning but I'm not 100% sure. Maybe this could be part of a "post-processing" class to unpack and further analyze/plot the data?